torch_xla/__init__.py

```diff
@@ -29,6 +29,14 @@ def _setup_xla_flags():
       flags, (('xla_gpu_simplify_all_fp_conversions', 'false'),))
   flags = _set_missing_flags(flags,
                              (('xla_gpu_force_compilation_parallelism', '8'),))
+  # This flag will rerun the latency hidding scheduler if the default
+  # shared memory limit 95% leads to OOM. Each rerun will choose a value
+  # 0.9x of the previous run, and the number of rerun is set to 1 now.
+  # Shared memory limit refers to --xla_tpu_scheduler_percent_shared_memory_limit.
+  # Lower shared memory limit means less communiation and computation overlapping,
+  # and thus worse performance.
+  flags = _set_missing_flags(flags,
+                             (('xla_latency_hiding_scheduler_rerun', '1'),))
```

Comment:
Nice! I was actually wondering about this, we set `XLA_FLAGS` here, but in other cases we pass the flags through `LIBTPU_INIT_ARGS`. Do you know if there's a difference? If you saw the appropriate log output from the test, it seems both work...

Reply:
oh right.. sorry I missed this. so for everything in `compiler/xla/xla.proto` we used `XLA_FLAGS`. `xla_latency_hiding_scheduler_rerun ` is one of those TPU specified flags that needs to be passed in with `LIBTPU_INIT_ARGS`.

Reply:
Good catch. I miss this lol.

Reply:
Ah, makes sense that `LIBTPU_INIT_ARGS` would be tpu-specific lol, thanks @JackCaoG. Is there a rule of thumb to tell which flag goes where? I'm thinking in terms of hashing the compilation environment, I suppose we'll just need to ensure both env vars are included.